modules/internal/ChapelDistribution.chpl

```diff
@@ -816,4 +890,71 @@ module ChapelDistribution {
     // runs the array destructor
     delete arr;
   }
+
+  // domain assigment helpers
+
```

Comment:
I wondered a bit about putting these routines into a module other than this one, possibly a new one, as this one has seemed poorly named and/or overloaded for too long now.  Ultimately, I think we should be splitting ChapelDistribution and ChapelArray into approximately a module per major class/record.  But it'd be helpful to not dive into that before we land array views so maybe we can park these here until then.

Should their names be chpl_ protected?  Or if they're intended for end-users, should we chpldoc them?